ark/segmentation/marker_quantification.py

```diff
@@ -263,6 +263,9 @@ def generate_cell_table(segmentation_labels, tiff_dir, img_sub_folder,
             filenames = io_utils.list_folders(tiff_dir)
             fovs = filenames
 
+    # ensures removal of file extensions, happens if fovs is not originally None
+    fovs = io_utils.extract_delimited_names(fovs, delimiter=None)
```

Comment:
Okay, I couldn't figure out why this worked, but then I saw that the "default" for delimiter is actually the default for the second split, not the first split. It first splits off anything with a period, and then applies the specified user delimiter. I think this is unnecessarily complicated, and means that people can never specify a custom delimiter that doens't split based on period. 

I think we should change the default of this function to be splitting on a period, and remove the double split call. 
Is there any reason this would mess things up?

Reply:
@ngreenwald I think the problem is that this function needs to handle both `fov1.tif` and `fov1_Row0Depth0....tif` types of files. The former doesn't need a double split, but the latter will.

I think we're better off setting the default delimiter to `None` because in that way, the user doesn't need to pass in any delimiter if they only want to split by period, but can additionally specify `_` if they need to. Also, we should change `delimiter` to something like `extra_delimiter` to avoid confusion. I don't see a use case where the user needs to specify multiple `extra_delimiters`, but that can be easily handled if we see this case pop up more often.

Reply:
If they have `fov1_Row0Depth0....tif`, won't it work if they supply that entire string, including .tiff, as delimiter?

Reply:
@ngreenwald So I think this basically affirms the fact that `extract_delimited_names` is really confusing.

Since this function is used only for extracting text from file names, I was under the assumption that by default, the user would want the extension (`.tif`, `.tiff`, etc.) removed. In other words, calling split('.')[0] is always necessary and we don't need an extra argument for that.

The `extra delimiter` argument would be if another delimiter is needed to extract just the `fovn` text from the file name. That would apply for a file name like `fov1_Row0Depth0....tif`.

Yes, you could supply the entire string, including `.tiff`, as delimiter and it would work, but that's not a very intuitive way of doing it. I feel like throughout the repo, we're setting some crazy stuff as delimiter and while it works, it leads to general confusion overall.

I feel like we should stick to passing single character delimiters to simplify our calls to `extract_delimited_names`. It would definitely help make it clearer what this function should be doing.

Reply:
Good point. I think the main use case Adam wrote this for originally was to simplify loading of MIBI-Tiffs in our repo, since they had that extra metadata. However, we're now using it for general extension cleanup as well. 

How about you create a separate function, `remove_file_extension`, that will take a list and remove .tiff, .tif, .jgp, etc from a list of names. Then we can remove the first split from `extract_delimited_names`, and call  `remove_file_extensions` everywhere in the repo where that's really the only thing we want

Reply:
@ngreenwald that sounds like a good idea. I'll adjust the PR description to include this cleanup task as well.